latest/docs-ref-autogen/reference-index.yml

```diff
@@ -118,7 +118,7 @@ directCommands:
     summary: Use CLI's old authentication flow based on device code. CLI will also use this if it can't launch a browser in your behalf, e.g. in remote SSH or Cloud Shell.
     description: ''
   - name: --username -u
-    summary: User name, service principal, or managed service identity ID.
+    summary: User principal name (`upn`); service principal or managed service Application identity ID (`appId`, not `objectId`).
```

Comment:
For user account, it doesn't need to be `userPrincipalName`. The `mail` `alias@microsoft.com` can be used to login to a guest tenant. In the guest tenant, `userPrincipalName` is actually like `alias_microsoft.com#EXT#@AzureSDKTeam.onmicrosoft.com` (got from `az ad signed-in-user show`).

Reply:
In order to use `mail`, it needs to be on a federated domain, right? Else you won't authenticate. So `mail` is probably not the safest nor bulletproof option.

Reply:
@juasli do you agree?